src/Symfony/Component/Console/Command/Command.php

```diff
@@ -250,12 +253,22 @@ public function run(InputInterface $input, OutputInterface $output)
 
         $input->validate();
 
-        if ($this->code) {
-            $statusCode = call_user_func($this->code, $input, $output);
-        } else {
-            $statusCode = $this->execute($input, $output);
+        try {
+            if (!$this->lock($output)) {
+                $statusCode = 0;
+            } elseif ($this->code) {
+                $statusCode = call_user_func($this->code, $input, $output);
+            } else {
+                $statusCode = $this->execute($input, $output);
+            }
+        } catch (\Exception $e) {
+            $this->release();
+
+            throw $e;
         }
 
+        $this->release();
```

Comment:
Hmm I can't find anything decent in the hhvm docs.. Maybe it's time to drop hhvm support as Symfony will probably never be a hhvm framework and hhvm seems to go a completely different way than php anyway.